apps/api/src/app/blueprint/e2e/create-blueprint.e2e.ts

```diff
@@ -0,0 +1,115 @@
+import { expect } from 'chai';
+
+import { UserSession } from '@novu/testing';
+import { EmailBlockTypeEnum, StepTypeEnum, INotificationTemplate, FilterPartTypeEnum } from '@novu/shared';
+import { NotificationTemplateRepository, EnvironmentRepository } from '@novu/dal';
+
+import { CreateNotificationTemplateRequestDto } from '../../notification-template/dto';
+
+describe('Create Notification template from blueprint - /blueprints/:templateId (POST)', async () => {
+  let session: UserSession;
+  const notificationTemplateRepository: NotificationTemplateRepository = new NotificationTemplateRepository();
+  const environmentRepository: EnvironmentRepository = new EnvironmentRepository();
+
+  before(async () => {
+    session = new UserSession();
+    await session.initialize();
+  });
+
+  it('should create template from blueprint', async function () {
+    const prodEnv = await getProductionEnvironment();
+
+    const { testTemplate, normalTemplate, prodVersionNotification, createdTemplate } =
```

Comment:
these are super confused I'm not following what you are trying to do here

Reply:
The entire test or those lines?
This is a legacy code that I refactored a bit in the following PRs, here I created createTemplateFromBlueprint in order to extract the logic of the creation of the template from the blueprint.

Reply:
I feel like we need to split this test into two separate, rn it checks if the blueprint will/won't be created, so these could be two tests.

Reply:
Good point, I will create a note for myself to revisit this test.